tests/integration/orcid/test_orcid_api.py

```diff
@@ -0,0 +1,135 @@
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""ORCID util tests."""
+
+from __future__ import absolute_import, division, print_function
+
+import os
+
+import pkg_resources
+import pytest
+import re
+import requests
+import requests_mock
+
+from inspirehep.modules.orcid.api import push_record_with_orcid, get_author_putcodes
+
+
+@pytest.fixture
+def mocked_internal_services(api_client):
+    hep_response = api_client.get('/literature/4328')
+    bibtex_response = api_client.get('/literature/4328', headers={
+        'Accept': 'application/x-bibtex',
+    })
+
+    def api_matcher(request):
+        if request.url == 'http://localhost:5000/api/literature/4328':
+            resp = requests.Response()
+            resp.status_code = 200
+            if request.headers['Accept'] == 'application/json':
+                resp._content = hep_response.data
+                return resp
+            elif request.headers['Accept'] == 'application/x-bibtex':
+                resp._content = bibtex_response.data
+                return resp
+
+    with requests_mock.Mocker() as requests_mocker:
+        requests_mocker.register_uri(
+            requests_mock.ANY,
+            re.compile('.*(indexer).*'),
+            real_http=True,
+        )
+        requests_mocker.add_matcher(api_matcher)
+        yield requests_mocker
+
+
+@pytest.fixture
+def mock_orcid_api(app, mocked_internal_services):
+    """Yield a MemberAPI and mock responses"""
+    mocked_internal_services.put(
+        'https://api.sandbox.orcid.org/v2.0/0000-0002-1825-0097/work/895497',
+        text='200 OK'
+    )
+    mocked_internal_services.post(
+        'https://api.sandbox.orcid.org/v2.0/0000-0002-1825-0097/work',
+        headers={
+            'Location': 'https://api.sandbox.orcid.org/v2.0/0000-0002-1825-0097/work/123456'
+        },
+    )
+    return mocked_internal_services
+
+
+def test_push_record_with_orcid(api_client, mock_orcid_api):
+    sample_orcid = '0000-0002-1825-0097'
+    expected_put_code = '895497'
+
+    result_put_code = push_record_with_orcid(
+        '4328', sample_orcid, 'fake-token', '895497'
+    )
+
+    assert expected_put_code == result_put_code
+
+
+def test_push_record_with_orcid_new(api_client, mock_orcid_api):
+    sample_orcid = '0000-0002-1825-0097'
+    expected_put_code = '123456'
+    result_put_code = push_record_with_orcid(
+        '4328', sample_orcid, 'fake-token', None
+    )
+
+    assert expected_put_code == result_put_code
+
+
+def get_file(fixture_name):
+    """Get contents of fixture files"""
+    path = pkg_resources.resource_filename(
+        __name__,
+        os.path.join('fixtures', fixture_name)
+    )
+    with open(path, 'r') as file:
+        return file.read()
```

Comment:
Don't use `file` as a var name, better use something like `fixture_fd` (for file descriptor). I think though that there's somewhere in the code something that already does this fixture retrieving thingie.

Also the name of the method is misleading, I'd expect `get_file` to get me a `file`, instead is returning it's contents.